monasca_events_api/tests/first_test.py

```diff
@@ -12,10 +12,228 @@
+        eventsObj = EventsSubClass()
+        eventsObj._events_repo = EventsRepository()
+        res = mock.MagicMock(spec='status')
+        res.body = {}
+        self.assertRaises(falcon.HTTPInternalServerError,
+                          eventsObj.on_get, self._generate_req(), res)
+        try:
+            eventsObj.on_get(self._generate_req(), res, None)
+        except Exception as e:
+            self.assertEqual(e.status, '500 Internal Server Error')
+
+    @mock.patch('monasca_events_api.common.messaging.kafka_publisher.KafkaPublisher')
+    @mock.patch('monasca_events_api.v2.common.helpers.validate_authorization')
+    @mock.patch('monasca_events_api.v2.events.Events._validate_event')
+    @mock.patch('monasca_events_api.v2.common.helpers.read_http_resource')
+    @mock.patch('monasca_events_api.v2.common.helpers.validate_json_content_type')
+    @mock.patch('monasca_events_api.v2.common.helpers.get_tenant_id')
+    def test_on_post_kafka_down(self, tenantid, json, http, event, validate, kafka):
+        """POST method with Kafka Down"""
+        kafka.send_message.side_effect = exceptions.MessageQueueException()
+
+        json.return_value = None
+        validate.return_value = True
+        http.return_value = self._generate_req()
+        tenantid.return_value = '0ab1ac0a-2867-402d'
+        event.return_value = True
+
+        eventsObj = EventsSubClass()
+        eventsObj._message_queue = kafka
+        res = mock.MagicMock()
+        res.body = {}
+        res.status = 0
+        self.assertRaises(falcon.HTTPInternalServerError,
+                          eventsObj.on_post, self._generate_req(), res)
+        try:
+            eventsObj.on_post(self._generate_req(), res)
+        except Exception as e:
+            self.assertEqual(e.status, '500 Internal Server Error')
+
+    @mock.patch('monasca_events_api.common.messaging.kafka_publisher.KafkaPublisher')
+    @mock.patch('monasca_events_api.v2.common.helpers.validate_authorization')
+    @mock.patch('monasca_events_api.v2.events.Events._validate_event')
+    @mock.patch('monasca_events_api.v2.common.helpers.read_http_resource')
+    @mock.patch('monasca_events_api.v2.common.helpers.validate_json_content_type')
+    @mock.patch('monasca_events_api.v2.common.helpers.get_tenant_id')
+    def test_on_post_pass(self, tenantid, json, http, event, validate, kafka):
+        """POST method success """
+        json.return_value = None
+        validate.return_value = True
+        http.return_value = True
+        tenantid.return_value = '0ab1ac0a-2867-402d'
+        event.return_value = True
+
+        eventsObj = EventsSubClass()
+        eventsObj._message_queue = kafka
+        res = mock.MagicMock()
+        res.body = {}
+        res.status = 0
+        try:
+            eventsObj.on_post(self._generate_req(), res)
+        except Exception:
+            self.fail("GET() raised HTTPInternalServerError unexpectedly")
```

Comment:
It's worth grouping all your tests for on method together.  It makes it easier to see if we've covered all the right cases.